packages/dom/src/dom/clean-node-list.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * External dependencies
+ */
+import { includes, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import isEmpty from './is-empty';
+import remove from './remove';
+import unwrap from './unwrap';
+import { isPhrasingContent } from '../phrasing-content';
+import insertAfter from './insert-after';
+
+/**
+ * Given a schema, unwraps or removes nodes, attributes and classes on a node
+ * list.
+ *
+ * @param {NodeList} nodeList The nodeList to filter.
+ * @param {Document} doc      The document of the nodeList.
+ * @param {Object}   schema   An array of functions that can mutate with the provided node.
+ * @param {Object}   inline   Whether to clean for inline mode.
+ */
+export default function cleanNodeList( nodeList, doc, schema, inline ) {
```

Comment:
this function is super complex when it comes to if-else nesting. Maybe it is possible to break it into smaller functions?

Reply:
That's definitely out of scope of this PR as it's only meant to split the existing module into smaller modules to make it easier to review incremental JSDoc typing, but might be a decent ticket to open and have a discussion about.